zircon.core/common/src/main/kotlin/org/hexworks/zircon/api/builder/component/HorizontalNumberInputBuilder.kt

```diff
@@ -0,0 +1,55 @@
+package org.hexworks.zircon.api.builder.component
+
+import org.hexworks.zircon.api.component.NumberInput
+import org.hexworks.zircon.api.component.base.BaseComponentBuilder
+import org.hexworks.zircon.api.component.data.CommonComponentProperties
+import org.hexworks.zircon.api.component.data.ComponentMetadata
+import org.hexworks.zircon.api.component.renderer.ComponentRenderer
+import org.hexworks.zircon.api.component.renderer.impl.DefaultComponentRenderingStrategy
+import org.hexworks.zircon.internal.component.impl.DefaultHorizontalNumberInput
+import org.hexworks.zircon.internal.component.renderer.DefaultNumberInputRenderer
+import kotlin.jvm.JvmStatic
+import kotlin.math.max
+
+@Suppress("UNCHECKED_CAST")
+data class HorizontalNumberInputBuilder(
+        val width: Int,
+        private var initialValue: Int = 0,
+        private var maxValue: Int = Int.MAX_VALUE,
```

Comment:
It might make sense to have a `minValue` if we have a `maxValue`.

Reply:
Yeah I wondered, the same way maybe the slider could specify a min and max value.

Reply:
Yep, makes sense!